### PR TITLE
Add build check

### PR DESCRIPTION
**Pull request by Kewlan**
This apparently works! 
Adds a check that tests if the pull request can be built successfully.

Same as clang check, if it fails click on `Details` and see what's wrong:
image
image

Requesting review from @PhlexPlexico.

### CODE REVIEW

**Comment by PhlexPlexico** [comment]
LGTM 👍 :)

File: .github/workflows/build_check.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: Build Check
+
+on:
+  pull_request:
+
+jobs:
+  build:
```
Comment (PhlexPlexico) [inline .github/workflows/build_check.yml]: One nitpick though - maybe just under `build` you could add name like `name: Compile Project` (or whatever you prefer, just something a bit more descriptive?) so it just doesn't have the single `build` when you click into the action?

image

Comment (Kewlan) [reply .github/workflows/build_check.yml]: Added names to steps, and combined the checks into one file!

**Review by PhlexPlexico — APPROVED** [review]
🔥

**Pull request merged**